src/base/CharacterSet.cc

```diff
@@ -168,3 +168,18 @@ CharacterSet::RFC3986_UNRESERVED()
     return *chars;
 }
 
+const CharacterSet &
+CharacterSet::libcSpace()
+{
+    const auto creator = []() -> auto {
+        const auto chars = new CharacterSet("libcSpace");
+        for (size_t idx = 0; idx < 256; ++idx) {
+            if (xisspace(idx))
```

Comment:
I added this CharacterSet to preserve locale-specific behavior of existing squid.conf parsing code (that used to call xisspace() via trim_trailing_ws(), FindStatement(), and when parsing `include` preprocessor directive). We may decide to stop being locale-sensitive in these parsing contexts (e.g., like we stopped relying on isspace() when validating Content-Length value in commit 9fcf5206), but let's not make that decision in _this_ behavior-preserving PR.

Reply:
Lets do make that decision before baking support for indeterminate invisible characters into squid.conf.
It is an easy one to make, 

FTR, the Squid community is **already** sharing `squid.conf` snippets widely with the assumption that the contents **do not** contain any locale-specific characters.

So ... please drop this logic. Only support the portable `CharacterSet::WSP` set for whitespace characters.

Reply:
> Lets do make that decision before baking support for indeterminate invisible characters into squid.conf.

The implied assumption that this PR is "baking support for indeterminate invisible characters into squid.conf" is false: This PR preserves official code behavior in this area. In other words, that (questionable) behavior is already as "baked in" as it will be after this PR. I strongly recommend changing this in a followup PR. Do you insist on _this_ PR changing Squid behavior in this area?

Reply:
Adding complicated code, knowing it will be erased in a (nearly immediate) followup? that is a bad practice @rousskov.

Just omitting the new complex code solves several issues and makes this PR a fair amount cleaner.

Reply:
> Adding complicated code, knowing it will be erased in a (nearly immediate) followup? that is a bad practice

This PR does not add complicated code. A bit of added code will probably be eventually erased, but that is not a big deal _and_ a followup PR (if done right) will have to modify a lot more code, code that this PR does not and should not touch.

> Just omitting the new complex code solves several issues and makes this PR a fair amount cleaner.

Not really. On the other hand, insisting on those out of scope changes (with significant side effects!) does delay this innocent PR even more. It is sad that straightforward, correct, useful, small, and functionality-preserving Squid changes are so costly!

Reply:
> please drop this logic. Only support the portable `CharacterSet::WSP` set for whitespace characters.

Done in #1858 to keep serious behavior changes (and associated noise) outside this "upgrade to SBuf" PR.

Reply:
> > please drop this logic. Only support the portable `CharacterSet::WSP` set for whitespace characters.
> 
> Done in #1858 to keep serious behavior changes (and associated noise) outside this "upgrade to SBuf" PR.

And those official changes are now merged into this PR branch. I have updated PR description accordingly.

When resolving conflicts, I found two more relevant space handling cases. One of them (former strwordtok() in `include` filename parsing) could have been handled in commit 314e430, but I missed it because it did not call isspace() directly. Another (parsing integers in `if` conditions) could not be handled earlier because strtoll() and friends are using isspace() implicitly (and we could not remove those calls without upgrading to Tokenizer which is done in this PR).

@yadij, I hope this master merge addresses your change request. Please review.